static/js/components/widgets/WidgetEditDialog_test.js

```diff
@@ -185,29 +188,47 @@ describe("WidgetEditDialog", () => {
         dialogData.validation.widget_type
       )
     })
+    ;[
+      [true, WIDGET_TYPE_SELECT, true],
+      [false, WIDGET_TYPE_SELECT, false],
+      [false, WIDGET_CREATE, true],
+      [false, WIDGET_CREATE, false],
+      [false, WIDGET_EDIT, true],
+      [false, WIDGET_EDIT, false]
+    ].forEach(([expected, dialogState, dialogOpen]) => {
+      it(`${shouldIf(
+        expected
+      )} shift focus from radio buttons when state=${dialogState} and dialog ${isIf(
+        dialogOpen
+      )} open`, () => {
+        dialogData.state = dialogState
 
-    it("disables automatic focus on radio buttons by focusing on the submit button", () => {
-      const div = document.createElement("div")
-      // $FlowFixMe: document.body should almost never be null
-      document.body.appendChild(div)
+        const div = document.createElement("div")
+        // $FlowFixMe: document.body should almost never be null
+        document.body.appendChild(div)
 
-      mount(
-        <WidgetEditDialog
-          dialogData={dialogData}
-          dialogOpen={true}
-          setDialogData={setDialogDataStub}
-          setDialogVisibility={setDialogVisibilityStub}
-          specs={specs}
-          updateForm={updateFormStub}
-        />,
-        {
-          attachTo: div
+        mount(
+          <Provider store={helper.store}>
```

Comment:
why is the `<Provider>` necessary now? it seems like you're still passing in the mocked `dialogData` instead of doing a big setup with dispatching actions to the store and all that

Reply:
It looks like it was always necessary. If I don't have it I see some randomly flaky tests with this error message:

     Invariant Violation: Could not find "store" in either the context or props of "Connect(Editor)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(Editor)".
      at invariant (node_modules/invariant/invariant.js:42:15)
      at new Connect (node_modules/react-redux/lib/components/connectAdvanced.js:134:33)
      at constructClassInstance (node_modules/react-dom/cjs/react-dom.development.js:11213:18)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:14337:5)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:15213:16)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:18780:12)
      at workLoop (node_modules/react-dom/cjs/react-dom.development.js:18820:24)
      at renderRoot (node_modules/react-dom/cjs/react-dom.development.js:18903:7)
      at performWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:19800:7)
      at performWork (node_modules/react-dom/cjs/react-dom.development.js:19712:7)
      at performSyncWork (node_modules/react-dom/cjs/react-dom.development.js:19686:3)
      at requestWork (node_modules/react-dom/cjs/react-dom.development.js:19555:5)
      at scheduleWork (node_modules/react-dom/cjs/react-dom.development.js:19364:5)
      at scheduleRootUpdate (node_modules/react-dom/cjs/react-dom.development.js:20032:3)
      at updateContainerAtExpirationTime (node_modules/react-dom/cjs/react-dom.development.js:20060:10)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:20117:10)
      at ReactRoot.render (node_modules/react-dom/cjs/react-dom.development.js:20413:3)
      at /src/node_modules/react-dom/cjs/react-dom.development.js:20550:14
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:19915:10)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:20546:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:20615:12)
      at Object.render (node_modules/enzyme-adapter-react-16/build/ReactSixteenAdapter.js:367:114)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:134:16)
      at mount (node_modules/enzyme/build/mount.js:21:10)
      at Context.<anonymous> (static/js/components/widgets/WidgetEditDialog_test.js:210:9)